### PR TITLE
Install page enhacement: version dropdown moved to PageHeader

### PR DESCRIPTION
**Pull request by heatlikeheatwave**
## 🔗 Relevant links

- [Asana task](https://app.asana.com/0/1204817897905865/1205712837415354/f) 🎟️
- [Figma](https://figmashort.link/GJTJF3)

## 🗒️ What

- Removed some dead code related to latest version as this is handled [here](https://github.com/hashicorp/dev-portal/blob/main/src/views/product-downloads-view/helpers.ts#L189-L204)

- Removed header subtitle:
Screenshot 2023-11-03 at 6 34 20 PM

- Moved `<VersionContextSwitcher />` to `PageHeader`

## 🤷 Why

For consistency, We'd like to update to the new component (version toggle/dropdown) used in our [documents](https://dev-portal-git-ksupdate-welcome-message-width-hashicorp.vercel.app/consul/docs) template and move it's placement to the page header.

### Mobile
Screenshot 2023-11-03 at 7 02 00 PM

### Desktop
Screenshot 2023-11-03 at 7 02 11 PM

## 🧪 Testing

- Navigate to [Vault install page](https://dev-portal-git-heat-featversion-select-update-hashicorp.vercel.app/vault/downloads)

### CODE REVIEW

File: src/components/version-context-switcher/index.tsx [hunk 1]
```diff
@@ -26,25 +27,8 @@ const VersionContextSwitcher = ({
...
-		const versionValue = options[i].value
-		if (versionLabel.indexOf('(latest)') >= 0) {
-			latestVersion = versionValue
-			break
-		}
-	}
-
-	const [selectedVersion, setSelectedVersion] = useState<
-		ContextSwitcherOption['value']
-	>(initialValue || latestVersion)
```
Comment (heatlikeheatwave) [inline src/components/version-context-switcher/index.tsx:47]: Handled here: https://github.com/hashicorp/dev-portal/blob/main/src/views/product-downloads-view/helpers.ts#L189-L204

File: src/components/version-context-switcher/index.tsx [hunk 2]
```diff
@@ -6,9 +6,10 @@
 import { ChangeEventHandler, ReactElement, useState } from 'react'
 import { IconCaret16 } from '@hashicorp/flight-icons/svg-react/caret-16'
 import { useCurrentProduct } from 'contexts'
-import ProductIcon from 'components/product-icon'
 import { ContextSwitcherOption, VersionContextSwitcherProps } from './types'
 import s from './version-context-switcher.module.css'
+import { ReleaseVersion } from 'lib/fetch-release-data'
+import { IconChevronDown16 } from '@hashicorp/flight-icons/svg-react/chevron-down-16'
```
Comment (kendallstrautman) [inline src/components/version-context-switcher/index.tsx:12]: ⛏️ nit: can you reorder these imports?

File: src/views/product-downloads-view/components/page-header/index.tsx [hunk 3]
```diff
@@ -52,10 +47,13 @@ const PageHeader = ({
 				>
 					{pageTitle}
 				</Heading>
-				<Text className={s.pageHeaderSubtitle} size={300} weight="regular">
-					{pageSubtitle}
-				</Text>
 			</div>
+			<VersionContextSwitcher
+				onChange={(e: SyntheticEvent) =>
+					setCurrentVersion(String(e.currentTarget))
```
Comment (kendallstrautman) [inline src/views/product-downloads-view/components/page-header/index.tsx:53]: Screenshot 2023-11-03 at 4 22 12 PM

Seeing an error on the preview when selecting a new version.

File: src/components/version-context-switcher/index.tsx [hunk 4]
```diff
@@ -58,30 +42,15 @@ const VersionContextSwitcher = ({
...
-	 * `ProductIcon` and the component intentionally returns `null` for if the
-	 * value for the `product` prop is `'sentinel'`, opting to pass a different
-	 * value for Sentinel here rather than affect all current references to
-	 * ProductIcon.
-	 *  - `ProductSwitcher` is one example that would be imacted by adding a
-	 *    ProductIcon for Sentinel. We currently do not want to render an icon
-	 *    for Sentinel in `ProductSwitcher`.
-	 */
-	const productIconSlug =
-		currentProduct.slug === 'sentinel' ? 'hcp' : currentProduct.slug
```
Comment (kendallstrautman) [inline src/components/version-context-switcher/index.tsx:72]: We will be supporting sentinel soon, so might be good to leave this here? Or any particular reason why you want to remove it with this work?

Comment (kellyschummer) [reply src/components/version-context-switcher/index.tsx:72]: agree- I don't think there's any reason to remove this.

File: src/components/version-context-switcher/index.tsx [hunk 5]
```diff
@@ -92,9 +61,7 @@ const VersionContextSwitcher = ({
 					</option>
 				))}
 			</select>
-			<span className={s.trailingIcon}>
-				<IconCaret16 />
-			</span>
+			<IconChevronDown16 className={s.trailingIcon} tab-index="-1" />
```
Comment (RubenSandwich) [inline src/components/version-context-switcher/index.tsx:64]: While I do appreciate the tab-index, i do not believe it is necessary. From my, so far limited and new, understanding of our codebase this chevron is this icon:

Screenshot showcasing what the author believes to be the chevron icon

Correct? Which if so then this Icon renders as an SVG, and SVGs are not natively marked as "interactive" so would not be in the "focus tree". So no tab-index would be required.

File: src/components/version-context-switcher/index.tsx (hunk 4 above)
Comment (heatlikeheatwave) [reply src/components/version-context-switcher/index.tsx:72]: I'm missing context here. Why would we need to keep this if we're getting rid of the product icon in the version dropdown?

Comment (heatlikeheatwave) [reply src/components/version-context-switcher/index.tsx:72]: @kendallstrautman @kellyschummer ☝️

File: src/views/product-downloads-view/components/page-header/index.tsx (hunk 3 above)
Comment (heatlikeheatwave) [reply src/views/product-downloads-view/components/page-header/index.tsx]: Fixed!

File: src/components/version-context-switcher/index.tsx (hunk 4 above)
Comment (kendallstrautman) [reply src/components/version-context-switcher/index.tsx:72]: Right! sorry I was wrong 👍🏻

**Review by kendallstrautman — APPROVED** [review]
Looks good from my end but would wait for @kellyschummer's ✅ to merge.

**Review by kellyschummer — APPROVED** [review]
LGTM- Thank you @heatlikeheatwave 🥳

**Pull request merged**